lib/iex/lib/iex/autocomplete.ex

```diff
@@ -28,8 +28,12 @@ defmodule IEx.Autocomplete do
 
   def expand([h | t] = expr, shell) do
     helper = get_helper(expr)
+    path = path_fragment(expr)
```

Comment:
We should probably try this only on the last clause of `cond`.

Reply:
Tested here, and it matches earlier on the cond clauses then. For example when I type `"./m<tab>` it would match for this clause: https://github.com/elixir-lang/elixir/pull/10895/files/213d13e29dba087ea2d4b9f3e444e1aa8ac9661c#diff-9e9de7d847547ae049e1297f35e4f7cd0f7d99d6f4cbcf8fa4b90c7c532ee035L45

Reply:
@kelvinst if I write this code:

```elixir
iex(1)> module = 123
123
iex(2)> "./mo
```

And hit tab, it won't auto-complete the variable. So it means we are ruling out it somewhere. I thought it was before we trigger this point?

Reply:
> So it means we are ruling out it somewhere.

I think it's discarded here: https://github.com/elixir-lang/elixir/blob/213d13e29dba087ea2d4b9f3e444e1aa8ac9661c/lib/iex/lib/iex/autocomplete.ex#L111-L133

So the cond actually matches opened strings like on your example, it just doesn't autocomplete because `Code.string_to_quoted(expr)` can't build the AST for it

Reply:
What we could do is to try a `Code.string_to_quoted(reduce(expr))` first, and only go to path completion if it fails. I think that would be the optimal way because most cond clauses do run some variation of that anyway.

Reply:
That way it might even make the opened string verification unnecessary, and we could just discard everything that's before the last `"`, removing all this: https://github.com/elixir-lang/elixir/blob/213d13e29dba087ea2d4b9f3e444e1aa8ac9661c/lib/iex/lib/iex/autocomplete.ex#L661-L673

Reply:
@josevalim I will test out this other way to implement, will let you know when I have something 👍